pandas/core/frame.py

```diff
@@ -7340,8 +7340,8 @@ def value_counts(
         >>> df
           first_name middle_name
         0       John       Smith
-        1       Anne        <NA>
```

Comment:
All the usages that were changed should still use `<NA>` rather than `NaN`. This should only change if `na_rep` was specified as something else

Reply:
> The expected value in these tests is <NA> but according to docs it should be NaN since that's the default value for na_rep. Somehow this behaviour is not in place.

@mroeschke

Reply:
I think since the current behavior has been the case for a few releases now, the default of `na_rep="NaN"` should be changed to `na_rep=lib.no_default` so that `NaN`, `NA`, and `None` can continue to use their reprs here

Reply:
@mroeschke the problem is if I set `na_rep=lib.no_default` then the string representation will not contain either of `NaN`, `NA` or `None`. I have tried setting the default as `<NA>` and reverted most of the test cases, however, there's issue in a couple of them.

Reply:
Using `na_rep=lib.no_default` would mean making `NaN`, `NA` and `None` return their string defaults like

```python
if self.na_rep is lib.no_default:
    ...existing logic
else:
    ...use na_rap
```